src/jsonLogic.js

```diff
@@ -212,3 +234,71 @@ function handleNestedObjectForComputedValues(values, formValues, parentID, valid
     })
   );
 }
+
+function buildSampleEmptyObject(schema = {}) {
+  const sample = {};
+  if (typeof schema !== 'object' || !schema.properties) {
+    return schema;
+  }
+
+  for (const key in schema.properties) {
+    if (schema.properties[key].type === 'object') {
+      sample[key] = buildSampleEmptyObject(schema.properties[key]);
+    } else if (schema.properties[key].type === 'array') {
+      const itemSchema = schema.properties[key].items;
+      sample[key] = buildSampleEmptyObject(itemSchema);
+    } else {
+      sample[key] = true;
+    }
+  }
+
+  return sample;
+}
+
+function validateInlineRules(jsonSchema, sampleEmptyObject) {
```

Comment:
Update: Another thing I remember while reviewing Part 5. This code assumes it will run/validate the json schema after it's created/written. But at Remote we'll build them in a UI (form builder). I wonder if these validations will run too soon. 

For example, I'm still creating the logic pointers (i.e. `x-jsf-logic-validations`) and it's already complaining that the validation declaration (`x-jsf-logic.validations`) doesn't exist. Or the other way around. I build the declaration but it says the field and or point doesn't exist yet. I think that's a very valid scenario to silent the checkers/validations while I'm in "edit mode". But not sure. We'd need to actually test it to see how it behaves.

Reply:
Yeah, I'm all up for a configuration and this could be a good idea - but lets see it when we feel the pain of using it internally.

Reply:
Agree with the sentiment of blocking the user in favor of more consistent data. I also think the internal UX of not running the validation while creating the schema is better, but that may be moot if the performance isn't an issue with re-running it constantly. Additionally, if performance or running the validation too frequently is an issue, we can optimize by debouncing, queuing, or possibly even off-loading some of the work to a service worker.

Reply:
Perfect! So If I understood correctly, let's leave it as it is, and then in the next iteration consider the config to toggle it off for dev.

Reply:
[Internal issue created](https://linear.app/remote/issue/WEB-45)